src/twisted/web/test/test_client.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Tests for various parts of L{twisted.web}.
+"""
+
+from twisted.trial import unittest
+
+from twisted.web import client
+
+class HTTP11FactoryTest(unittest.TestCase):
+    """
+    Unit tests for L{client._HTTP11ClientFactory}.
+    """
+
+    def test_repr(self):
+        def a_special_name():
+            return 1
+        # Call it for coverage
+        self.assertEquals(a_special_name(), 1)
+        factory = client._HTTP11ClientFactory(a_special_name,
+                                              'this_is_kinda_unique')
+        result = repr(factory)
+        self.assertIn('a_special_name', result)
+        self.assertIn('this_is_kinda_unique', result)
```

Comment:
This test doesn't seem to demonstrate any particularly compelling user-facing behavior.  No one is going to be instantiating an `_HTTP11ClientFactory` with these custom objects and then expecting to see them in the repr.

So, kind of repeating myself, what's useful about this behavior?